### ISSUE TITLE
An error message is not Displayable in ansible

### ISSUE THREAD

**Issue by BouchardClaude** [aristanetworks/ansible-avd#2572]
### Issue Summary

Using EOS_DESIGN 3.8.2 and having error in the Network Service / Tenants yml configs, some error are not displayed in Ansible configured with all requirement for 3.8.2.  Error is not happening in 3.7. Ansible show this error  :

```
TASK [arista.avd.eos_designs : Generate device configuration in structured format] ***
[WARNING]: Failure using method (v2_runner_on_failed) in callback plugin
(<ansible.plugins.callback.awx_display.CallbackModule object at
0x7f7f51f01430>): Object of type AristaAvdError is not JSON serializable
```

Look like the error message is not proper json?

### Which component(s) of AVD impacted

eos_designs

### How do you run AVD ?

Ansible AWX

### Relevant log output

```shell
TASK [arista.avd.eos_designs : Generate device configuration in structured format] ***
[WARNING]: Failure using method (v2_runner_on_failed) in callback plugin
(<ansible.plugins.callback.awx_display.CallbackModule object at
0x7f7f51f01430>): Object of type AristaAvdError is not JSON serializable
```
```


### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

**Comment by ClausHolbechArista** [comment]
We are not able to recreate this error without more context. Could you try to run your playbook with maximum verbosity and provide the output?

**Comment by carlbuchmann** [comment]
@philippebureau - can you help capture more information on this issue?

**Comment by BouchardClaude** [comment]
Hi, please see verbose log of the error on
 Anisble core 2.12.10
python version = 3.8.14
```
TASK [arista.avd.eos_designs : Generate device configuration in structured format] ***
task path: /tmp/bwrap_5143841_htyptanp/awx_5143841_tkyts0tw/requirements_collections/ansible_collections/arista/avd/roles/eos_designs/tasks/main.yml:51
[WARNING]: Failure using method (v2_runner_on_failed) in callback plugin
(<ansible.plugins.callback.awx_display.CallbackModule object at
0x7f7e7b52eca0>): Object of type AristaAvdError is not JSON serializable
Callback Exception: 
  File "/var/lib/awx/venv/NetOps-38/lib/python3.8/site-packages/ansible/executor/task_queue_manager.py", line 450, in send_callback
    method(*new_args, **kwargs)
   File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/ansible_runner/display_callback/module.py", line 397, in v2_runner_on_failed
    with self.capture_event_data('runner_on_failed', **event_data):
   File "/opt/rh/rh-python38/root/usr/lib64/python3.8/contextlib.py", line 113, in __enter__
    return next(self.gen)
   File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/ansible_runner/display_callback/module.py", line 100, in capture_event_data
    event_context.dump_begin(sys.stdout)
   File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/ansible_runner/display_callback/events.py", line 196, in dump_begin
    self.cache.set(":1:ev-{}".format(begin_dict['uuid']), begin_dict)
   File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/ansible_runner/display_callback/events.py", line 71, in set
    partial_data = json.dumps(value, cls=AnsibleJSONEncoderLocal)
   File "/opt/rh/rh-python38/root/usr/lib64/python3.8/json/__init__.py", line 234, in dumps
    return cls(
   File "/opt/rh/rh-python38/root/usr/lib64/python3.8/json/encoder.py", line 199, in encode
    chunks = self.iterencode(o, _one_shot=True)
   File "/opt/rh/rh-python38/root/usr/lib64/python3.8/json/encoder.py", line 257, in iterencode
    return _iterencode(o, 0)
   File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/ansible_runner/display_callback/events.py", line 51, in default
    return super(AnsibleJSONEncoderLocal, self).default(o)
   File "/opt/rh/rh-python38/root/usr/lib64/python3.8/json/encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
Callback Exception: 
  File "/var/lib/awx/venv/NetOps-38/lib/python3.8/site-packages/ansible/executor/task_queue_manager.py", line 450, in send_callback
    method(*new_args, **kwargs)
   File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/ansible_runner/display_callback/module.py", line 397, in v2_runner_on_failed
    with self.capture_event_data('runner_on_failed', **event_data):
   File "/opt/rh/rh-python38/root/usr/lib64/python3.8/contextlib.py", line 113, in __enter__
    return next(self.gen)
   File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/ansible_runner/display_callback/module.py", line 100, in capture_event_data
    event_context.dump_begin(sys.stdout)
   File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/ansible_runner/display_callback/events.py", line 196, in dump_begin
    self.cache.set(":1:ev-{}".format(begin_dict['uuid']), begin_dict)
   File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/ansible_runner/display_callback/events.py", line 71, in set
    partial_data = json.dumps(value, cls=AnsibleJSONEncoderLocal)
   File "/opt/rh/rh-python38/root/usr/lib64/python3.8/json/__init__.py", line 234, in dumps
    return cls(
   File "/opt/rh/rh-python38/root/usr/lib64/python3.8/json/encoder.py", line 199, in encode
    chunks = self.iterencode(o, _one_shot=True)
   File "/opt/rh/rh-python38/root/usr/lib64/python3.8/json/encoder.py", line 257, in iterencode
    return _iterencode(o, 0)
   File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/ansible_runner/display_callback/events.py", line 51, in default
    return super(AnsibleJSONEncoderLocal, self).default(o)
   File "/opt/rh/rh-python38/root/usr/lib64/python3.8/json/encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
Callback Exception: 
  File "/var/lib/awx/venv/NetOps-38/lib/python3.8/site-packages/ansible/executor/task_queue_manager.py", line 450, in send_callback
    method(*new_args, **kwargs)
   File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/ansible_runner/display_callback/module.py", line 397, in v2_runner_on_failed
    with self.capture_event_data('runner_on_failed', **event_data):
   File "/opt/rh/rh-python38/root/usr/lib64/python3.8/contextlib.py", line 113, in __enter__
    return next(self.gen)
   File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/ansible_runner/display_callback/module.py", line 100, in capture_event_data
    event_context.dump_begin(sys.stdout)
   File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/ansible_runner/display_callback/events.py", line 196, in dump_begin
    self.cache.set(":1:ev-{}".format(begin_dict['uuid']), begin_dict)
   File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/ansible_runner/display_callback/events.py", line 71, in set
    partial_data = json.dumps(value, cls=AnsibleJSONEncoderLocal)
   File "/opt/rh/rh-python38/root/usr/lib64/python3.8/json/__init__.py", line 234, in dumps
    return cls(
   File "/opt/rh/rh-python38/root/usr/lib64/python3.8/json/encoder.py", line 199, in encode
    chunks = self.iterencode(o, _one_shot=True)
   File "/opt/rh/rh-python38/root/usr/lib64/python3.8/json/encoder.py", line 257, in iterencode
    return _iterencode(o, 0)
   File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/ansible_runner/display_callback/events.py", line 51, in default
    return super(AnsibleJSONEncoderLocal, self).default(o)
   File "/opt/rh/rh-python38/root/usr/lib64/python3.8/json/encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
Callback Exception: 
  File "/var/lib/awx/venv/NetOps-38/lib/python3.8/site-packages/ansible/executor/task_queue_manager.py", line 450, in send_callback
    method(*new_args, **kwargs)
   File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/ansible_runner/display_callback/module.py", line 397, in v2_runner_on_failed
    with self.capture_event_data('runner_on_failed', **event_data):
   File "/opt/rh/rh-python38/root/usr/lib64/python3.8/contextlib.py", line 113, in __enter__
    return next(self.gen)
   File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/ansible_runner/display_callback/module.py", line 100, in capture_event_data
    event_context.dump_begin(sys.stdout)
   File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/ansible_runner/display_callback/events.py", line 196, in dump_begin
    self.cache.set(":1:ev-{}".format(begin_dict['uuid']), begin_dict)
   File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/ansible_runner/display_callback/events.py", line 71, in set
    partial_data = json.dumps(value, cls=AnsibleJSONEncoderLocal)
   File "/opt/rh/rh-python38/root/usr/lib64/python3.8/json/__init__.py", line 234, in dumps
    return cls(
   File "/opt/rh/rh-python38/root/usr/lib64/python3.8/json/encoder.py", line 199, in encode
    chunks = self.iterencode(o, _one_shot=True)
   File "/opt/rh/rh-python38/root/usr/lib64/python3.8/json/encoder.py", line 257, in iterencode
    return _iterencode(o, 0)
   File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/ansible_runner/display_callback/events.py", line 51, in default
    return super(AnsibleJSONEncoderLocal, self).default(o)
   File "/opt/rh/rh-python38/root/usr/lib64/python3.8/json/encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '

```

**Comment by carlbuchmann** [comment]
@BouchardClaude - thanks for the additional details... I see two versions of Python being leveraged here, 3.6 and 3.8, and ansible-core 2.12.10 only supports 3.8 on the control node -> https://docs.ansible.com/ansible/latest/installation_guide/intro_installation.html

Can you reproduce the issue outside of AWX, in a Linux virtual environment with all our latest requirements installed -> https://github.com/aristanetworks/ansible-avd/blob/devel/ansible_collections/arista/avd/requirements.txt ?

**Comment by BouchardClaude** [comment]
Hi, 
I jsut tried with a fresh clean install with python 3.8 and only Anisble core 2.12.10 and go the same error using the demo project ( as expexted). 
I did also confirm that the error is when I dont put l3_interfaces nodes in brackets [  ] . ( I expect it to not work as it expect brackets, but im also expecting to print the error so it point to what portion of the yml is wrong.

Using max verbose, im able to see the error, but I dont feel it should do a traceback ( maybe the brackets are breaking the json? ) :

`
TASK [arista.avd.eos_designs : Generate device configuration in structured format] *****************************************************************************************************************
task path: /home/bouchardclaude/.ansible/collections/ansible_collections/arista/avd/roles/eos_designs/tasks/main.yml:51
<localhost> ESTABLISH LOCAL CONNECTION FOR USER: bouchardclaude
<localhost> EXEC /bin/sh -c 'echo ~bouchardclaude && sleep 0'
<localhost> ESTABLISH LOCAL CONNECTION FOR USER: bouchardclaude
<localhost> EXEC /bin/sh -c 'echo ~bouchardclaude && sleep 0'
<localhost> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/bouchardclaude/.ansible/tmp `"&& mkdir "` echo /home/bouchardclaude/.ansible/tmp/ansible-tmp-1677356164.9452994-13514-47710661664109 `" && echo ansible-tmp-1677356164.9452994-13514-47710661664109="` echo /home/bouchardclaude/.ansible/tmp/ansible-tmp-1677356164.9452994-13514-47710661664109 `" ) && sleep 0'
<localhost> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/bouchardclaude/.ansible/tmp `"&& mkdir "` echo /home/bouchardclaude/.ansible/tmp/ansible-tmp-1677356164.96235-13515-157406412057584 `" && echo ansible-tmp-1677356164.96235-13515-157406412057584="` echo /home/bouchardclaude/.ansible/tmp/ansible-tmp-1677356164.96235-13515-157406412057584 `" ) && sleep 0'
The full traceback is:
Traceback (most recent call last):
  File "/home/bouchardclaude/.ansible/collections/ansible_collections/arista/avd/plugins/action/yaml_templates_to_facts.py", line 150, in run
    template_result_data = cls_instance.render()
  File "/home/bouchardclaude/.ansible/collections/ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/avdstructuredconfig.py", line 80, in render
    return super().render()
  File "/home/bouchardclaude/.ansible/collections/ansible_collections/arista/avd/plugins/plugin_utils/avdfacts.py", line 61, in render
    return {key: getattr(self, key) for key in self.keys() if getattr(self, key) is not None}
  File "/home/bouchardclaude/.ansible/collections/ansible_collections/arista/avd/plugins/plugin_utils/avdfacts.py", line 61, in <dictcomp>
    return {key: getattr(self, key) for key in self.keys() if getattr(self, key) is not None}
  File "/usr/lib/python3.8/functools.py", line 967, in __get__
    val = self.func(instance)
  File "/home/bouchardclaude/.ansible/collections/ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/ethernet_interfaces.py", line 46, in ethernet_interfaces
    raise AristaAvdError(
ansible_collections.arista.avd.plugins.plugin_utils.errors.errors.AristaAvdError: Length of lists 'interfaces', 'nodes', 'ip_addresses' and 'descriptions' (if used) must match for l3_interfaces for VRF10 in TENANT1
[WARNING]: Failure using method (v2_runner_on_failed) in callback plugin (<ansible.plugins.callback.default.CallbackModule object at 0x7f0f4c785f70>): Object of type AristaAvdError is not JSON
serializable
Callback Exception:
  File "/home/bouchardclaude/.local/lib/python3.8/site-packages/ansible/executor/task_queue_manager.py", line 450, in send_callback
    method(*new_args, **kwargs)
   File "/home/bouchardclaude/.local/lib/python3.8/site-packages/ansible/plugins/callback/default.py", line 96, in v2_runner_on_failed
    msg = "fatal: [%s]: FAILED! => %s" % (host_label, self._dump_results(result._result))
   File "/home/bouchardclaude/.local/lib/python3.8/site-packages/ansible/plugins/callback/__init__.py", line 139, in _dump_results
    jsonified_results = json.dumps(OrderedDict(sorted(abridged_result.items(), key=to_text)),
   File "/usr/lib/python3.8/json/__init__.py", line 234, in dumps
    return cls(
   File "/usr/lib/python3.8/json/encoder.py", line 201, in encode
    chunks = list(chunks)
   File "/usr/lib/python3.8/json/encoder.py", line 431, in _iterencode
    yield from _iterencode_dict(o, _current_indent_level)
   File "/usr/lib/python3.8/json/encoder.py", line 405, in _iterencode_dict
    yield from chunks
   File "/usr/lib/python3.8/json/encoder.py", line 438, in _iterencode
    o = _default(o)
   File "/home/bouchardclaude/.local/lib/python3.8/site-packages/ansible/module_utils/common/json.py", line 71, in default
    value = super(AnsibleJSONEncoder, self).default(o)
   File "/usr/lib/python3.8/json/encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
`

**Comment by ClausHolbechArista** [comment]
Thank you @BouchardClaude for reporting and helping us understand the issue in detail.

I have submitted a fix in #2578, so we return a proper error message instead of a traceback. The full traceback info is still available with -vvv to assist with further debugging.
